onnxruntime/core/session/inference_session.h

```diff
@@ -416,7 +418,16 @@ class InferenceSession {
 
   ModelMetadata model_metadata_;
   std::unordered_set<std::string> required_inputs_;
-  std::unordered_map<std::string, const NodeArg*> input_def_map_;
+
+  struct InputDefMetaData {
```

Comment:
don't see much usage of this newly added struct in this PR.

Reply:
This struct is used to cache the static values and extensively used inside inference_session.cc. Check validateinputs function. Reference to 'iter->second' is referring to this struct.